### PR TITLE
Clipping

### PR DESCRIPTION
**Pull request by possan**
Added simple canvas clipping support, and a slime cube example ...

### CODE REVIEW

**Comment by bit101** [comment]
I like this a lot. I just want to avoid properties that are arrays, because arrays are used to create animations between values. I know I used it for lineDash, and... maybe something else? But I'd like to avoid it.

My idea is to create a glc.rect function, so you could say

clipRect: glc.rect(50, 50, 100, 100),


Then you could potentially do something like 

clipRect: [glc.rect(50, 50, 100, 100), glc.rect(100, 100, 200, 200)],

to animate it. I'll play around with it, but I'm in the middle of a bunch of other stuff so may not get to it right away. But thanks.

**Comment by tpltnt** [comment]
What is the state of this PR?

File: sketches/examples/slimecube.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+			lineWidth: 4,
+			clip: [0, 2 * i, 200, 2],
+		shadowColor: '#fff',
+		shadowOffsetX: 0.0,
+		shadowOffsetY: 0.0,
+		shadowBlur: 20.0,
+		});
+	}
+
+}
```
Comment (vitorgalvao) [inline sketches/examples/slimecube.js]: It has a “no newline at end of file” warning. Those are [actually meaningful](http://unix.stackexchange.com/a/18789).

File: sketches/glc-min.js [hunk 2]
```diff
@@ -25,4 +25,4 @@
...
...cksettings"],function(e){function a(a,f){n=a,r=f,t=e.create(n.w+220,20,"Output"),t.setWidth(n.w+12),t.addImage(i,""),t.addInfo(s,""),t.addButton(o,r.clearOutput),t.addButton(u,function(){p(),t.hide()})}function f(e){t.show();var n="data:image/gif;base64,"+d(e);t.setImageURL(i,n);var r="data:image/gif;base64,",o=Math.round((n.length-r.length)*3/4);t.setInfo(s,"Approx size: "+Math.round(o/1024)+"kb")}function l(e){t.show(),t.setImageURL(i,e);var n="data:image/png;base64,",r=Math.round((e.length-n.length)*3/4);t.setInfo(s,"Approx size: "+Math.round(r/1024)+"kb")}function c(e){t.setWidth(e)}function h(e,n){t.setPosition(e,n)}function p(){t.setImageURL(i,""),t.setInfo(s,""),t.setWidth(n.w+12)}function d(e){var t="",n=0,r=e.length,i="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=",s,o,u,a,f,l,c;while(n<r)s=e.charCodeAt(n++),o=e.charCodeAt(n++),u=e.charCodeAt(n++),a=s>>2,f=(s&3)<<4|o>>4,l=(o&15)<<2|u>>6,c=u&63,isNaN(o)?l=c=64:isNaN(u)&&(c=64),t=t+i.charAt(a)+i.charAt(f)+i.charAt(l)+i.charAt(c);return t}var t=null,n=null,r=null,i="Capture",s="size",o="Clear Image",u="Close";return{init:a,setGIF:f,setPNG:l,setWidth:c,setPosition:h,clearOutput:p}}),define("app/glc",["app/r
```
Comment (vitorgalvao) [inline sketches/glc-min.js]: Same (“no newline at end of file” warning).

File: glc/glc-min.js [hunk 3]
```diff
@@ -25,4 +25,4 @@
...
...cksettings"],function(e){function a(a,f){n=a,r=f,t=e.create(n.w+220,20,"Output"),t.setWidth(n.w+12),t.addImage(i,""),t.addInfo(s,""),t.addButton(o,r.clearOutput),t.addButton(u,function(){p(),t.hide()})}function f(e){t.show();var n="data:image/gif;base64,"+d(e);t.setImageURL(i,n);var r="data:image/gif;base64,",o=Math.round((n.length-r.length)*3/4);t.setInfo(s,"Approx size: "+Math.round(o/1024)+"kb")}function l(e){t.show(),t.setImageURL(i,e);var n="data:image/png;base64,",r=Math.round((e.length-n.length)*3/4);t.setInfo(s,"Approx size: "+Math.round(r/1024)+"kb")}function c(e){t.setWidth(e)}function h(e,n){t.setPosition(e,n)}function p(){t.setImageURL(i,""),t.setInfo(s,""),t.setWidth(n.w+12)}function d(e){var t="",n=0,r=e.length,i="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=",s,o,u,a,f,l,c;while(n<r)s=e.charCodeAt(n++),o=e.charCodeAt(n++),u=e.charCodeAt(n++),a=s>>2,f=(s&3)<<4|o>>4,l=(o&15)<<2|u>>6,c=u&63,isNaN(o)?l=c=64:isNaN(u)&&(c=64),t=t+i.charAt(a)+i.charAt(f)+i.charAt(l)+i.charAt(c);return t}var t=null,n=null,r=null,i="Capture",s="size",o="Clear Image",u="Close";return{init:a,setGIF:f,setPNG:l,setWidth:c,setPosition:h,clearOutput:p}}),define("app/glc",["app/r
```
Comment (vitorgalvao) [inline glc/glc-min.js]: Same (“no newline at end of file” warning).

**Pull request closed without merging**